src/mobile/src/main/java/tupperdate/android/home/Home.kt

```diff
@@ -1,98 +1,146 @@
+to call like() and dislike() without the weird RealRecipeApi import.
+ */
 fun Home(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile?,
-    modifier: Modifier = Modifier,
+    recipeApi: RecipeApi,
+    onChatClick: () -> Unit,
+    onProfileClick: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
-    if (user == null) {
-        HomeDisconnected(
-            onButtonClick,
-            modifier,
-        )
-    } else {
-        HomeConnected(
-            onButtonClick,
-            user,
-            modifier,
-        )
-    }
-}
+    // TODO make a call to front element of recipeApi
+    val presentRecipe = (RecipeApi.Recipe(
+        "Red lobster",
+        "Look at me, am I not yummy ?",
+        "lobster.jpg"
+    ))
 
-@Composable
-private fun HomeDisconnected(
-    onButtonClick: () -> Unit,
-    modifier: Modifier,
-) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there.")
-        Button(onClick = onButtonClick) {
-            Text("Go to onboarding")
-        }
-    }
+    mainTopBar(onChatClick = onChatClick, onProfileClick = onProfileClick)
+
+    DisplayRecipeCard(
+        presentRecipe = presentRecipe,
+        onLike = { recipeApi.like(presentRecipe) },
+        onDislike = { recipeApi.dislike(presentRecipe) },
+        onReturnClick = onReturnClick,
+        onRecipeClick = onRecipeClick
+    )
 }
 
 @Composable
-private fun HomeConnected(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun DisplayRecipeCard(
+    presentRecipe: RecipeApi.Recipe,
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there ${user.displayName}")
-        Button(onClick = onButtonClick) {
-            Text("View Branding")
+    Row(Modifier.fillMaxSize()) {
+        Column(
+            Modifier.fillMaxWidth().align(Alignment.Bottom)
+                .padding(bottom = veryLittleButtonSize.dp)
+        ) {
+            WithConstraints { -> //why no parameter here ??
+
+                //determine height and width of card according to constraints
+                //the card measures 90 % of screen width and 75 % of screen height
+                val boxHeight =
+                    with(DensityAmbient.current) { constraints.maxHeight.toDp() * 0.75f }
+                val boxWidth = with(DensityAmbient.current) { constraints.maxWidth.toDp() * 0.9f }
+
+                //the align(..) doesn't work anymore (because of reasons), so we need to
+                //set manually a default pos
+                val defaultPos = ((with(DensityAmbient.current) { constraints.maxWidth.toDp() / 2 })
+                        - boxWidth / 2).value.toInt()
+                val (pos, setPos) = remember { mutableStateOf((defaultPos)) }
```

Comment:
This is because you're in a `WithConstraints` scope. It might be worth doing things a bit differently; you probably do not need absolute positioning of your composable.

Reply:
In particular, the swipe slop does not have to be a percentage of the card width, but probably a fixed size (think `160.dp`). If the user is in landscape (or even worse, on a phablet), he should have to scroll the same amount to dismiss the card, not 10x more because the screen is 10x bigger.